patches/node/fix_comment_out_incompatible_crypto_modules.patch

```diff
@@ -8,86 +8,288 @@ and https://github.com/nodejs/node/pull/31178 that is not currently compatible
++++ b/src/crypto/crypto_rsa.cc
+@@ -210,10 +210,12 @@ WebCryptoCipherStatus RSA_Cipher(
+   if (label_len > 0) {
+     void* label = OPENSSL_memdup(params.label.get(), label_len);
+     CHECK_NOT_NULL(label);
++#ifndef OPENSSL_IS_BORINGSSL
+     if (EVP_PKEY_CTX_set0_rsa_oaep_label(ctx.get(), label, label_len) <= 0) {
+       OPENSSL_free(label);
+       return WebCryptoCipherStatus::FAILED;
+     }
++#endif
+   }
+ 
+   size_t out_len = 0;
+diff --git a/src/crypto/crypto_util.cc b/src/crypto/crypto_util.cc
+index fbe4c0f06c10df7c8a492711594381c27c6f81f7..76c8c037ffd3c8b67179d7d881ad6ea530b00686 100644
+--- a/src/crypto/crypto_util.cc
++++ b/src/crypto/crypto_util.cc
+@@ -20,6 +20,8 @@
+ #endif
+ #endif
+ 
++#include <openssl/rand.h>
++
+ namespace node {
+ 
+ using v8::ArrayBuffer;
+@@ -118,7 +120,6 @@ void InitCryptoOnce() {
+   OPENSSL_init_ssl(0, settings);
+   OPENSSL_INIT_free(settings);
+   settings = nullptr;
+-#endif
+ 
+ #ifndef _WIN32
+   if (per_process::cli_options->secure_heap != 0) {
+@@ -137,6 +138,7 @@ void InitCryptoOnce() {
+         break;
+     }
+   }
 +#endif
-       EVP_PKEY* raw_params = nullptr;
-       if (EVP_PKEY_paramgen(param_ctx.get(), &raw_params) <= 0)
-         return nullptr;
+ #endif
+ 
+   /* Override FIPS settings in cnf file, if needed. */
+@@ -675,10 +677,12 @@ void SecureBuffer(const FunctionCallbackInfo<Value>& args) {
+ }
+ 
+ void SecureHeapUsed(const FunctionCallbackInfo<Value>& args) {
++#ifndef OPENSSL_IS_BORINGSSL
+   Environment* env = Environment::GetCurrent(args);
+   if (CRYPTO_secure_malloc_initialized())
+     args.GetReturnValue().Set(
+         BigInt::New(env->isolate(), CRYPTO_secure_used()));
++#endif
+ }
+ }  // namespace
+ 
+diff --git a/src/crypto/crypto_util.h b/src/crypto/crypto_util.h
+index f2f61aa45185812e9248845b664539be4fe24550..303ba4c3b7c4c2fc5dee906e22d5e7642b8351c8 100644
+--- a/src/crypto/crypto_util.h
++++ b/src/crypto/crypto_util.h
+@@ -15,7 +15,9 @@
+ #include <openssl/err.h>
+ #include <openssl/evp.h>
+ #include <openssl/ec.h>
++#ifndef OPENSSL_IS_BORINGSSL
+ #include <openssl/kdf.h>
++#endif
+ #include <openssl/rsa.h>
+ #include <openssl/dsa.h>
+ #include <openssl/ssl.h>
+diff --git a/src/node.cc b/src/node.cc
+index a99f2f4f01b16fc70f3038a25da22c788714b3f6..ceb532648b33a6c2f3b8a135b315985cfff5419e 100644
+--- a/src/node.cc
++++ b/src/node.cc
+@@ -1020,7 +1020,7 @@ InitializationResult InitializeOncePerProcess(int argc, char** argv) {
+     return result;
+   }
+ 
+-#if HAVE_OPENSSL
++#if HAVE_OPENSSL && !defined(OPENSSL_IS_BORINGSSL)
+   {
+     std::string extra_ca_certs;
+     if (credentials::SafeGetenv("NODE_EXTRA_CA_CERTS", &extra_ca_certs))
+diff --git a/src/node_metadata.h b/src/node_metadata.h
+index 4486d5af2c1622c7c8f44401dc3ebb986d8e3c2e..db1769f1b3f1617ed8dbbea57b5e324183b42be2 100644
+--- a/src/node_metadata.h
++++ b/src/node_metadata.h
+@@ -6,7 +6,7 @@
+ #include <string>
+ #include "node_version.h"
+ 
+-#if HAVE_OPENSSL
++#if 0
```

Comment:
what does this fix?

Reply:
kept getting this weird error:
```
../../third_party/electron_node/src/node_metadata.h:10:10: fatal error: 'openssl/crypto.h' file not found
#include <openssl/crypto.h>
         ^~~~~~~~~~~~~~~~~~
1 error generated.
```

so i added it as a stopgap before i could look into it further